App.tsx

```diff
@@ -1,59 +1,50 @@
-import {
-  NavigationContainer,
-  type NavigationProp,
-} from '@react-navigation/native';
-import {createNativeStackNavigator} from '@react-navigation/native-stack';
-import {useColorScheme} from 'react-native';
+import React, {useState, useEffect} from 'react';
+import {View, Text} from 'react-native';
+import auth, {type FirebaseAuthTypes} from '@react-native-firebase/auth';
+import {NavigationContainer} from '@react-navigation/native';
 
-import {FavouritesProvider} from './providers/FavouritesProvider';
-import HomeScreen from './screens/HomeScreen';
-import DetailedMuseumScreen from './screens/DetailedMuseumScreen';
-import type Museum from './model/Museum';
-import {getAppColors} from './utils/colors';
-import {getAppStyles} from './utils/styles';
+import UnAuthNavigator from './navigators/UnAuthNavigator';
+import AuthNavigator from './navigators/AuthNavigator';
+import {AuthProviders} from './providers/AuthProviders';
 
-export type MainRootStackParamList = {
-  Home: undefined;
-  DetailedMuseum: {museum: Museum};
-};
+type NullableUser = FirebaseAuthTypes.User | null;
 
-export type MainStackNavigation = NavigationProp<MainRootStackParamList>;
+export default function App() {
+  // Set an initializing state whilst Firebase connects
+  const [initializing, setInitializing] = useState(true);
+  const [user, setUser] = useState<NullableUser>();
 
-const Stack = createNativeStackNavigator<MainRootStackParamList>();
+  // Handle user state changes
+  function onAuthStateChanged(user: NullableUser) {
+    setUser(user);
+    if (initializing) setInitializing(false);
```

Comment:
It's not a mistake or anything, but I always recommend wrapping if statements in curly braces, because it makes the code a bit easier to read, especially in more complex functions 😄 
```suggestion
    if (initializing) {
      setInitializing(false);
    }
```

Reply:
imo if instruction is short there is no need to change it

Reply:
As I've said, it's not a mistake and I can live with that, it's more my personal preference, because I've caught myself misreading the code a couple of times 😄